src/store/slices/testingHistory.slice.ts

```diff
@@ -24,7 +24,8 @@ const initialState: TestingHistoryState = {
 
 export const fetchHistory = createAsyncThunk("fetchHistory", async (payload: {}, thunkApi) => {
   try {
-    const response = await fetch<Run[]>(thunkApi, { method: 'GET', url: '/run' });
+    const apiUrl = payload ? `/profile/${payload}/runs`: '/run';
```

Comment:
Same here, we can avoid the `profileId` as a parameter and obtain it directly from the state